schematic/models/validate_attribute.py

```diff
@@ -1769,123 +1834,111 @@ def _get_rule_scope(self, val_rule: str) -> ScopeTypes:
         scope = val_rule.lower().split(" ")[2]
         return scope
 
+    def _get_target_manifest_dataframes(
+        self,
+        target_component: str,
+        project_scope: Optional[list[str]] = None,
+        access_token: Optional[str] = None,
+    ) -> dict[str, pd.DataFrame]:
+        """Returns target manifest dataframes in the form of a dictionary
+
+        Args:
+            target_component (str): The component to get manifests for
+            project_scope (Optional[list[str]], optional):
+             Projects to limit the scope of cross manifest validation to. Defaults to None.
+            access_token (Optional[str], optional): Asset Store access token. Defaults to None.
+
+        Returns:
+            dict[str, pd.DataFrame]: Keys are synapse ids, values are datframes of the synapse id
+        """
+        manifest_ids, dataset_ids = self.get_target_manifests(
+            target_component, project_scope, access_token
+        )
+        manifests: list[pd.DataFrame] = []
+        for dataset_id in dataset_ids:
+            entity: File = self.synStore.getDatasetManifest(
+                datasetId=dataset_id, downloadFile=True
+            )
+            manifests.append(pd.read_csv(entity.path))
```

Comment:
It would be a larger change, but this might be a good opportunity to introduce some concurrency here. The benefit is that you would be able to download all of the datasets in parallel rather than one-by-one.

You have a known list of `dataset_ids` at this point that you need to call `getDatasetManifest` for each of them. You could do something to what @linglp did in https://github.com/Sage-Bionetworks/schematic/blob/fds-305-table-operations/schematic/store/synapse.py#L1787-L1825

I can help you write this code if you wanted to pair program this feature out @andrewelamb . This would also be the start of using the new `File` model I wrote as well that you'd be able to use for it's async interface: https://python-docs.synapse.org/reference/oop/models_async/#synapseclient.models.File